plugin/keyring_vault/vault_keyring.cc

```diff
@@ -0,0 +1,198 @@
+#include <my_global.h>
+#include <mysql/plugin_keyring.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include "keyring.h"
+#include "vault_keys_container.h"
+#include "vault_parser.h"
+#include "vault_io.h"
+
+using keyring::IVault_curl;
+using keyring::IVault_parser;
+using keyring::Vault_parser;
+using keyring::Vault_io;
+using keyring::Vault_keys_container;
+using keyring::Vault_curl;
+using keyring::Logger;
+
+int check_keyring_file_data(MYSQL_THD thd  MY_ATTRIBUTE((unused)),
+                            struct st_mysql_sys_var *var  MY_ATTRIBUTE((unused)),
+                            void *save, st_mysql_value *value)
+{
+  char            buff[FN_REFLEN+1];
+  const char      *keyring_filename;
+  int             len = sizeof(buff);
```

Comment:
len is letter used in calling value->val_str(value, buff, &len), which takes int* as argument. I could change len to size_t, but only to cast it to int two 3 lines below.